rasa/nlu/test.py

```diff
@@ -898,6 +912,79 @@ def compute_metrics(
     return (intent_metrics, entity_metrics, intent_results, entity_results)
 
 
+def compare_nlu(
+    configs: List[Text],
+    data: TrainingData,
+    exclusion_percentages: List[int],
+    micros: Dict[Text, Any],
+    model_names: List[Text],
+    output: Text,
+    runs: int,
+) -> List[int]:
+    intent_examples_present = []
+
+    for run in range(runs):
+
+        logger.info("Beginning comparison run {}/{}".format(run + 1, runs))
+        train, test = data.train_test_split()
+
+        run_path = os.path.join(output, "run_{}".format(run + 1))
+        create_path(run_path)
+
+        test_path = os.path.join(run_path, "test.md")
+        create_path(test_path)
+
+        write_to_file(test_path, test.as_markdown())
+        intent_examples_present = []
+
+        for percentage in exclusion_percentages:
+            percent_string = "{}%_exclusion".format(percentage)
+
+            _, train = train.train_test_split(percentage / 100)
+            intent_examples_present.append(len(train.training_examples))
+
+            out_path = os.path.join(run_path, percent_string)
+            train_split_path = os.path.join(out_path, "train.md")
+            create_path(train_split_path)
+
+            write_to_file(train_split_path, train.as_markdown())
+
+            for nlu_config, model_name in zip(configs, model_names):
+
+                logger.info(
+                    "Evaluating config '{}' with {}".format(model_name, percent_string)
+                )
+
+                try:
+                    model_path = train_nlu(
```

Comment:
do we have to persist the data?

Can we do?

```
trainer = Trainer(config.load(config_file))
    trainer.train(training_data)
trainer.persist(model_path, fixed_model_name="nlu")
```

Reply:
The code is used somewhere inside `train_nlu`. Using `train_nlu` has the benefit that the model is already zipped in the end.